packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -0,0 +1,500 @@
+      color: CupertinoColors.transparent,
+      child: _coverSheetSecondaryTransition(
+        widget.secondaryRouteAnimation,
+        _coverSheetPrimaryTransition(
+          context,
+          widget.primaryRouteAnimation,
+          ClipRRect(
+            borderRadius: const BorderRadius.vertical(top: Radius.circular(12)),
+            child: widget.child,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+/// Route for displaying an iOS sheet styled page.
+///
+/// The `CupertinoSheetRoute` will slide up from the bottom of the screen and stop
+/// below the top of the screen. If the previous route is a non-sheet route, than
+/// it will animate downwards to stack behind the new sheet. If the previous route
+/// is a sheet route, than it will animate slightly upwards to look like it is laying
+/// on top of the previous stack of sheets.
+///
+/// Typically called by [showCupertinoSheet], which provides some boilerplate for
+/// pushing the `CupertinoSheetRoute` to the root navigator and providing simple
+/// nested naviagation.
+///
+/// The sheet will be dismissed by dragging downwards on the screen, or a call to
+/// [CupertinoSheetRoute.popSheet].
+///
+/// {@tool dartpad}
+/// This example shows how to navigate to [CupertinoSheetRoute] by using it the
+/// same as a regular route.
+///
+/// ** See code in examples/api/lib/cupertino/sheet/cupertino_sheet.0.dart **
+/// {@end-tool}
+///
+/// {@tool dartpad}
+/// This example shows how to setup a [CupertinoSheetRoute] with nested navigation.
+/// This allows for pushing to new routes that transition from within the sheet
+/// view.
+///
+/// This example can be acieved by calling [showCupertinoSheet], which sets up
+/// much of the same boilderplate by default.
+///
+/// ** See code in examples/api/lib/cupertino/sheet/cupertino_sheet.1.dart **
+/// {@end-tool}
+class CupertinoSheetRoute<T> extends PageRoute<T> with CupertinoSheetRouteTransitionMixin<T> {
+  /// Creates a page route that displays an iOS styled sheet.
+  CupertinoSheetRoute({
+    required this.builder,
+  });
+
+  /// Builds the primary contents of the sheet route.
+  final WidgetBuilder builder;
+
+  @override
+  DelegatedTransitionBuilder? get delegatedTransition => CupertinoSheetTransition.delegateTransition;
+
+  @override
+  Widget buildContent(BuildContext context) {
+    return _CupertinoSheetScope(
+      child: builder(context),
+    );
+  }
+
+  /// Checks if a Cupertino sheet view exists in the widget tree above the current
+  /// context.
+  static bool hasParentSheet(BuildContext context) {
+    return _CupertinoSheetScope.maybeOf(context) != null;
+  }
+
+  /// Pops the entire [CupertinoSheetRoute], if a sheet route exists in the stack.
+  ///
+  /// Used if to pop an entire sheet at once, if there is nested naviagtion within
+  /// that sheet.
+  static void popSheet(BuildContext context) {
+    if (hasParentSheet(context)) {
+      Navigator.of(context, rootNavigator: true).pop();
```

Comment:
If we provide useRootNavigator in showCupertinoSheet this will also need to be updated